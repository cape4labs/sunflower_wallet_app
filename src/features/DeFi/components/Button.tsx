import { Pressable } from 'react-native';
import TextWithFont from '../../../shared/components/TextWithFont';

type ButtonType = {
  onPress: () => void;
  text: string;
  customStyle?: string;
  accent?: boolean;
};

// A general button used in this feature
export function Button({ onPress, text, customStyle, accent = false }: ButtonType) {
  return (
    <Pressable
      onPress={onPress}
      className={`justify-center items-center w-[90%] h-14 border-4 border-custom_border rounded-lg ${accent ? 'bg-custom_accent' : 'bg-custom_complement'} ${customStyle ?? ''}`}
    >
      <TextWithFont customStyle={`text-lg ${accent ? 'text-black' : 'text-white'}`}>{text}</TextWithFont>
    </Pressable>
  );
}
