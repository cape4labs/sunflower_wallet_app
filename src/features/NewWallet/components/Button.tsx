import { Pressable, Text } from 'react-native';

type ButtonType = {
  onPress: () => void;
  text: string;
};

// A general button used in this feature
export function Button({ onPress, text }: ButtonType) {
  return (
    <Pressable
      onPress={onPress}
      className="justify-center items-center bg-black w-full h-24 text-xl m-auto"
    >
      <Text className="text-white">{text}</Text>
    </Pressable>
  );
}
