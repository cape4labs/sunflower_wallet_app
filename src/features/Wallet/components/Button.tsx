import { Pressable, Text } from 'react-native';

type ButtonType = {
  onPress: () => void;
  text: string;
  customStyle?: string;
  accent?: boolean;
};

// A general button used in this feature
export function Button({ onPress, text, customStyle, accent = false }: ButtonType) {
  return (
    <Pressable
      onPress={onPress}
      className={`justify-center items-center w-[90%] h-14 border-4 border-custom_border rounded-lg ${accent ? 'bg-custom_accent' : 'bg-custom_complement'} ${customStyle ?? ''}`}
    >
      <Text className={`text-lg ${accent ? 'text-black' : 'text-white'}`}>{text}</Text>
    </Pressable>
  );
}
