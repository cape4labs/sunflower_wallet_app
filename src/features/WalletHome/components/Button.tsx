import { Pressable, Text } from 'react-native';
import { Send, Upload } from 'lucide-react-native';

type ButtonType = {
  onPress?: () => void;
  text: string;
  customStyle?: string;
  accent?: boolean;
  disable?: boolean;
  iconName: 'Send' | 'Upload'; 
};

type TextButtonType = {
  onPress?: () => void;
  text: string;
  customStyle?: string;
  accent?: boolean;
};

export function Button({
  onPress,
  text,
  customStyle,
  accent = false,
  disable = false,
  iconName,
}: ButtonType) {
  const IconComponent = iconName === 'Send' ? Send : Upload;

  return (
    <Pressable
      onPress={onPress}
      className={`justify-center items-center py-3 px-12 border-2 flex-row gap-3 border-custom_border rounded-2xl relative overflow-hidden ${
        accent ? 'bg-custom_accent' : 'bg-custom_complement'
      } ${customStyle ?? ''}`}
      disabled={disable}
    >
      <Text className={`text-lg ${accent ? 'text-black' : 'text-white'} z-10`}>{text}</Text>
      <IconComponent
        color={accent ? 'black' : 'white'}
        size={20}
        strokeWidth={1.5}
      />
    </Pressable>
  );
}

export function TextButton({ onPress, text, customStyle, accent = false }: TextButtonType) {
  return (
    <Pressable
      onPress={onPress}
      className={`justify-center items-center py-3 px-8 border-b-2 flex-row overflow-hidden ${
        accent ? 'border-white' : 'border-gray-500'
      } ${customStyle ?? ''}`}
    >
      <Text className={`text-xl ${accent ? 'text-custom_accent' : 'text-white'} z-10`}>
        {text}
      </Text>
    </Pressable>
  );
}