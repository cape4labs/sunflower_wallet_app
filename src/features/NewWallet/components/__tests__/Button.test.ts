import { render } from '@testing-library/react-native';
import { Button } from '../Button.tsx';

it('Button snapshot', () => {
  const tree = render(Button({ onPress: () => {}, text: 'test' })).toJSON();
  expect(tree).toMatchSnapshot();
});
