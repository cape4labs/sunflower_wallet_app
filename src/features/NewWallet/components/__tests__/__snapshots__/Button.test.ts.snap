// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  className="justify-center items-center py-3 px-14 border-[6px] border-custom_border rounded-2xl relative overflow-hidden bg-custom_complement "
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    className="font-[spacegrotesk] font-normal text-lg text-white z-10"
  >
    test
  </Text>
</View>
`;
